plugins/ua_plugin_securitypolicy.c

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2018 (c) Mark Giraud, Fraunhofer IOSB
+ */
+
+#include "ua_plugin_securitypolicy.h"
+
+size_t
+UA_SecurityPolicy_getRemoteAsymEncryptionBufferLengthOverhead(const UA_SecurityPolicy *securityPolicy,
```

Comment:
Hmm. Now we have a hard binding between a plugin and the main library.
This implementation is not intended to be exchanged.

I think it is better to stay with a per-policy implementation as before.

Reply:
Then every policy has to implement the same function. I don't think that's a good idea.
The implementation is in a plugin source which can be used by all other policies.

Reply:
> Now we have a hard binding between a plugin and the main library.

I assumed this method is also called from the main library.
Since you put it into a public header where it can be seen by the plugins and by the main lib.
But it is only used by the plugins.

How about we add a file `/plugins/ua_securitypolicy_helper.h` (or similar) that collects common functionality.
So we don't increase the API surface of the library but still reuse code between plugins.

Reply:
That's also possible. But the function is used by the main library and the plugins (in securechannel.c and in basic128rsa15.c)

Reply:
Right. I found out why I was confused.
I have to click "load" on securechannel.c to see the full diff on github.
Otherwise ctrl-c ctrl-f doesn't find the usage.

I don't care much about which way this goes.
But .. there should be a clear separation between the plugins and the main lib.

The plugins are meant to be exchangeable with no hard dependencies from the main lib.
Any interaction between code in /src and /plugins must go through a public plugin interface.

Reply:
Isn't that the case right now? The function is part of the public plugin api but has only one implementation in a general plugin source file.
Another approach would be to put it into the src part and make it available as part of the public api.

Imo the current solution should be sufficient and avoids having extra files that only contain a few lines of code. I think we could maybe refactor one or two functions that currently reside in the securechannel into the security policy just like the getOverhead function. I'll have to check though.

Reply:
> I think we could maybe refactor one or two functions that currently reside in the securechannel into the security policy just like the getOverhead function. I'll have to check though.

👍